### ISSUE TITLE
Mis-scope of let-binding declarations immediately followed by signatures

### ISSUE THREAD

**Issue by cigsender** [JustusAdam/language-haskell#215]
The first local signature (for `x :: ()`) in the image below should have the same scope as the signature for `y` below it:

![foo](https://user-images.githubusercontent.com/18196871/168651929-0c64cd79-5146-462e-aa76-0e87882c5d52.png)

**Comment by JustusAdam** [comment]
Yeah it should, but the VSCode grammar infrastructure doesn't have parser capabilities that are powerful enough to distinguish the two cases. 
I can elaborate more in case you want to take a stab at it, but as far as I am concerned this won't (be able to) get fixed.

**Comment by KristianBalaj** [comment]
+1
